mesh/src/test/java/no/nordicsemi/android/mesh/sensorutils/DevicePropertyCharacteristicTest.java

```diff
@@ -180,4 +180,47 @@ public void testDateUTC() {
             counter++;
         }
     }
+
+    @Test
+    public void energy32() {
+        final ArrayList<Double> expectedSamples = new ArrayList<>();
+        expectedSamples.add(0.0d);
+        expectedSamples.add(0.001d);
+        expectedSamples.add(2147483.646d);
+        expectedSamples.add(4294967.293d);
+        expectedSamples.add(null);
+        expectedSamples.add(null);
+        final ArrayList<byte[]> samples = new ArrayList<>();
+        samples.add(new byte[]{0x00, 0x00, 0x00, 0x00});
+        samples.add(new byte[]{0x01, 0x00, 0x00, 0x00});
+        samples.add(new byte[]{(byte) 0xFE, (byte) 0xFF, (byte) 0xFF, 0x7F});
```

Comment:
Any reason why you only cast some of the values?

Reply:
It's about how Java only supporting signed primitive number data types.
```
    byte[] bytes = {
      (byte) 0b11111111, // without (byte) this is interpreted as a signed two's complement 32-bit int with ones everywhere and doesn't fit in a byte, so compiler error.
      0b01111111, // fits in a byte, compiles fine.
    };
```